app/services/dataworks_mappers/open_alex.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+module DataworksMappers
+  # Map from OpenAlex metadata to Dataworks metadata
+  class OpenAlex < Base # rubocop:disable Metrics/ClassLength
+    IDENTIFIER_TYPES = {
+      'openalex' => 'OpenAlex',
+      'doi' => 'DOI'
+    }.freeze
+
+    def perform_map
+      {
+        identifiers:,
+        titles:,
+        creators:,
+        publication_year: source[:publication_year].to_s,
+        url: source.dig(:primary_location, :landing_page_url),
+        access:,
+        provider: 'OpenAlex',
+        dates:,
+        language: source[:language],
+        rights_list:,
+        subjects:,
+        funding_references:,
+        related_identifiers:
+      }.compact_blank
+    end
+
+    private
+
+    def access
+      return 'Restricted' unless source.dig(:primary_location, :is_oa)
+
```

Comment:
Not going to have this stand in the way of the PR but wondering whether this should be open_access at the top level instead of primary_location.  Works either way.

Reply:
Updated.